drivers/wifi/nrf700x/src/fmac_main.c

```diff
@@ -341,6 +341,27 @@ int nrf_wifi_reg_domain(const struct device *dev, struct wifi_reg_domain *reg_do
 	}
 
 	if (reg_domain->oper == WIFI_MGMT_SET) {
+#ifndef CONFIG_NRF700X_RADIO_TEST
+#ifdef CONFIG_NRF700X_STA_MODE
+		/* Need to check if WPA supplicant is initialized or not.
+		 * Must be cheecked when CONFIG_WPA_SUPP is enabled.
+		 * Not applicable for RADIO_TEST or when CONFIG_WPA_SUPP is not enabled.
+		 */
+		/* It is possbile that during supplicant initialization driver may
+		 * get the command. lock will try to ensure that supplicant
+		 * initialization is complete.
+		 */
+		k_mutex_lock(&vif_ctx_zep->vif_lock, K_FOREVER);
+
+		if (!vif_ctx_zep->supp_drv_if_ctx) {
```

Comment:
This check won't be enough, as this is set in the `wpa_drv_init` but the NM iface is registered after `add_interface`.

Reply:
You can use `wifi_nm_get_instance_iface` to see if the iface is managed by WPA supplicant or not. But those APIs don't have lock, so, might have to add one.

Reply:
Added required checks.